vault/resource_database_secret_backend_role.go

```diff
@@ -20,121 +24,132 @@ var (
 	databaseSecretBackendRoleNameFromPathRegex    = regexp.MustCompile("^.+/roles/(.+$)")
 )
 
+var roleAPIFields = []string{
+	consts.FieldDefaultTTL,
+	consts.FieldMaxTTL,
+	consts.FieldCreationStatements,
+	consts.FieldRevocationStatements,
+	consts.FieldRollbackStatements,
+	consts.FieldRenewStatements,
+	consts.FieldCredentialConfig,
```

Comment:
What is the expected behavior when TFVP uses a credential config for type `client_certificate` for Vault versions that don't support that type? I suspect databaseSecretBackendRoleRead may hit errors in this case. It doesn't look like type `client_certificate` is available for Vault 1.13 and below: https://developer.hashicorp.com/vault/api-docs/v1.13.x/secret/databases#credential_config

Reply:
For versions below Vault 1.11, if the user tries to use the credential_type we would return an error message that it is not supported.